### ISSUE TITLE
Type analysis: Boolean is not detected on ternary operator

### ISSUE THREAD

**Issue by RafaelSalguero** [Microsoft/TypeScript#14950]
The "boolean" type is not detected as an option on a cascaded ternary operator
**TypeScript Version:**  2.2.6

**Code**
```ts
//This code does not work
//Property 'toISOString' does not exist on type 'Boolean | Date'.
var parameter: string | number | Date | Boolean;
        const str =
            (typeof (parameter) == "number") ? "" + parameter :
                (typeof (parameter) == "boolean") ? "" + parameter : 
                    (typeof (parameter) == "string") ? parameter :
                        (parameter.toISOString) ? parameter.toISOString() : ""; 

//This code do work:
var parameter: string | number | Date;
  const str =
            (typeof (parameter) == "number") ? "" + parameter :
                    (typeof (parameter) == "string") ? parameter :
                        (parameter.toISOString) ? parameter.toISOString() : "";

```
**Expected **
If the condition `typeof(parameter) == "boolean"` is not meet, the parameter type should be inferred to be `Date`

**Actual behavior:**
The type is inferred to be `Date | Boolean`

**Comment by RyanCavanaugh** [comment]
The correct type name is `boolean`, not `Boolean`. See http://stackoverflow.com/questions/14727044/typescript-difference-between-string-and-string (same thing applies to `boolean` / `Boolean`)
